### PR TITLE
Fix for isStable loading for quick order

### LINKED ISSUE #13282: Quick order infinite loop when cart is not stable
**Describe the bug**
When cart is not stable, quick order has an infinite spinner

**To Reproduce**
1. clear all session/local storage AND hard refresh
3. click quick order as anonymous user from the top nav bar selectors
4. see infinite spinner

**Expected behavior**
Should show quick order page with the form, list, and actions

**Screenshots/Video**

https://user-images.githubusercontent.com/29099466/127381058-b2e4dd32-48a5-407e-848b-32e39e582845.mov

### PR DESCRIPTION
**Pull request by dydome**
close #13282

### CODE REVIEW

File: feature-libs/cart/quick-order/components/quick-order/quick-order.component.ts [hunk 1]
```diff
@@ -9,21 +14,26 @@ import {
...
 })
-export class QuickOrderComponent implements OnInit {
+export class QuickOrderComponent implements OnInit, OnDestroy {
   cartId$: Observable<string>;
   entries$: Observable<OrderEntry[]>;
-  isCartLoading$: Observable<boolean> = this.activeCartService.isStable();
   quickOrderListLimit$: Observable<
     number | undefined
   > = this.component.data$.pipe(map((data) => data.quickOrderListLimit));
+  isCartStable$: Observable<any> = combineLatest([
```
Comment (bgambocjaviniar) [inline feature-libs/cart/quick-order/components/quick-order/quick-order.component.ts:31]: Let's talk after/before standup

I'm not quite sure why we need a spinner for your page as it's not dependent on any data especially with cart.

Comment (bgambocjaviniar) [inline feature-libs/cart/quick-order/components/quick-order/quick-order.component.ts]:
```suggestion
  isCartStable$: Observable<boolean> = combineLatest([
```

However, let's talk about this, or maybe you can bring this up to someone as I know GC will complain once he sees this flicker.

Example of the flicker @ 19s and 34s (flicker = the cart became stable and before the items were added to the cart, therefore we see the page for a split sec or so)


https://user-images.githubusercontent.com/29099466/127596047-30c191ef-d7f2-4bfc-b620-7059ed9bd590.mov

Comment (bgambocjaviniar) [reply feature-libs/cart/quick-order/components/quick-order/quick-order.component.ts]: You can try hooking into  `CartAddEntryEvent` for load and remove the spinner when `CartEntryAddEntrySuccess/FailEvent` when you press Add To Cart (primary button)

File: feature-libs/cart/quick-order/components/quick-order/form/quick-order-form.component.ts [hunk 2]
```diff
@@ -36,8 +36,18 @@ export class QuickOrderFormComponent implements OnInit, OnDestroy {
     this.validateProductControl(value);
   }
 
+  get loading(): boolean {
```
Comment (bgambocjaviniar) [inline feature-libs/cart/quick-order/components/quick-order/form/quick-order-form.component.ts]: we have one above called `isDisabled`, so let's name this `isLoading` to stay consistent. Update instances throughout the PR as well.

File: feature-libs/cart/quick-order/components/quick-order/form/quick-order-form.component.ts [hunk 3]
```diff
@@ -36,8 +36,18 @@ export class QuickOrderFormComponent implements OnInit, OnDestroy {
...
+    return this._loading;
+  }
+
+  @Input('loading') set loading(value: boolean) {
+    this._loading = value;
```
Comment (bgambocjaviniar) [inline feature-libs/cart/quick-order/components/quick-order/form/quick-order-form.component.ts:44]: add unit test for this to see if the input is disabled similarly to `isDisabled` unit test.

File: feature-libs/cart/quick-order/components/quick-order/quick-order.component.ts [hunk 4]
```diff
@@ -9,21 +14,26 @@ import {
...
 })
-export class QuickOrderComponent implements OnInit {
+export class QuickOrderComponent implements OnInit, OnDestroy {
   cartId$: Observable<string>;
   entries$: Observable<OrderEntry[]>;
-  isCartStable$: Observable<boolean> = this.activeCartService.isStable();
   quickOrderListLimit$: Observable<
     number | undefined
   > = this.component.data$.pipe(map((data) => data.quickOrderListLimit));
+  isCartStable$: Observable<boolean> = combineLatest([
```
Comment (bgambocjaviniar) [inline feature-libs/cart/quick-order/components/quick-order/quick-order.component.ts:31]: add unit test for this to determine the output of each emission

**Review by bgambocjaviniar — CHANGES_REQUESTED** [review]
LGTM, just some naming differences + needing unit test (someone can approve during your time if you'd like)

However, as long as the team agrees about disabling most of the links / actions on the quick order page, then I'm all for it. Reason why I'm asking is I see we're disabling links to the product and such, but we can always navigate out of this page by clicking a category, homepage, my-account, etc.

**Review by mszczepaniak — APPROVED** [review]

**Pull request merged**